openstack/utils/templates/snippets/_set_ingress_cors_annotations.tpl

```diff
@@ -0,0 +1,10 @@
+{{ define "utils.snippets.set_ingress_cors_annotations" }}
+{{- if .Values.cors.enabled }}
```

Comment:
I would suggest to change this to 
```
{{- if and .Values.cors.enabled .Values.utils.cors.allow_origin }}
```
as I think it probably doesn't make much sense to set the annotations without `allow_origin` set

Reply:
I think that's confusing to the user. The user would expect CORS to be enabled, but it isn't because of a misconfiguration (on their side?). Instead, I'd rather add a `required` to using the `allow_origin` - what do you think?

Reply:
True, that sounds better.